vortex-array/src/array/list/mod.rs

```diff
@@ -0,0 +1,288 @@
+mod compute;
+
+use std::fmt::Display;
+use std::sync::Arc;
+
+use num_traits::AsPrimitive;
+use serde::{Deserialize, Serialize};
+use vortex_dtype::{match_each_native_ptype, DType};
+use vortex_error::{vortex_bail, vortex_panic, VortexExpect, VortexResult};
+
+use crate::array::{NullArray, PrimitiveArray};
+use crate::compute::{scalar_at, slice};
+use crate::encoding::ids;
+use crate::stats::{ArrayStatistics, Stat, StatisticsVTable, StatsSet};
+use crate::validity::{LogicalValidity, Validity, ValidityMetadata, ValidityVTable};
+use crate::variants::{ListArrayTrait, PrimitiveArrayTrait, VariantsVTable};
+use crate::visitor::{ArrayVisitor, VisitorVTable};
+use crate::{
+    impl_encoding, ArrayDType, ArrayData, ArrayLen, ArrayTrait, Canonical, IntoArrayData,
+    IntoArrayVariant, IntoCanonical,
+};
+
+impl_encoding!("vortex.list", ids::LIST, List);
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ListMetadata {
+    validity: ValidityMetadata,
+    element_len: usize,
+    offset_dtype: DType,
```

Comment:
Can this be PType?

Reply:
We want to store the nullability, so I would say the dtype still preferred?

Reply:
I thought you assert that offsets are non-nullable?